File: py/number-complement.py
class Solution(object):
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        n = num
        n |= (n >>  1)
        n |= (n >>  2)
        n |= (n >>  4)
        n |= (n >>  8)
        n |= (n >> 16)
        return n ^ num

File: go/number-complement.go
package leetcode

func findComplement(num int) int {
	x := num
	x |= x >> 1
	x |= x >> 2
	x |= x >> 4
	x |= x >> 8
	x |= x >> 16
	x >>= 1
	return ^num & x
}
